drydock/templates/drydock/k8s/lifecycle/cms.yml

```diff
@@ -7,21 +7,41 @@ spec:
     spec:
       containers:
         - name: cms
-          lifecycle:
-            preStop:
-              exec:
-                command:
-                - "sleep"
-                - "30"
           startupProbe:
             httpGet:
               httpHeaders:
                 - name: Host
                   value: {{ CMS_HOST }}
               path: /heartbeat
               port: 8000
-            initialDelaySeconds: 5
+            initialDelaySeconds: 1
             timeoutSeconds: 3
-            periodSeconds: 5
-            failureThreshold: 5
+            periodSeconds: 1
+            failureThreshold: 30
+          livenessProbe:
+            httpGet:
+              httpHeaders:
+                - name: Host
+                  value: {{ LMS_HOST }}
+              path: /heartbeat
+              port: 8000
+            initialDelaySeconds: 3
+            timeoutSeconds: 30
+            periodSeconds: 60
+            failureThreshold: 2
+            successThreshold: 1
       terminationGracePeriodSeconds: 60
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: cms
+              topologyKey: kubernetes.io/hostname
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0
+      maxSurge: 2
```

Comment:
Wouldn't this make the update really slow if we have, say, 100 pods or something like that? Shouldn't it be a percentage?

Reply:
Yes, the default value is 25% which seems good for us